### PR TITLE
Reimplemented method cell_face_as_dense in the grid class

### PR DESCRIPTION
**Pull request by keileg**
## Proposed changes
Changes:
1. Replaced the (8 year old!) implementation of `cell_face_as_dense` in the `Grid` class. The new implementation is much more transparent, and documented.
2. Renamed the method `cell_face_as_dense` -> `cell_faces_as_dense`.
3. Improved the test for this method.

@IvarStefansson @IngridKJ: Although the test suite is much improved, the old implementation did not fail on this expanded set. I am not sure which grid you used to make the method fail today, but could you test the new version and see if it works? If so, I am not sure if we should expand the test suite with the previously failing grid, so let me know what you think. If it still fails, we will definitely add it.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Testing (contribution related to testing of existing or new functionality).
- [x] Documentation (contribution related to adding, improving, or fixing documentation).
- [x] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: src/porepy/grids/grid.py
Comment (IvarStefansson) [inline src/porepy/grids/grid.py:928]:
```suggestion
        # Find the non-zero elements in the cell-face relation.
```

**Review by IvarStefansson — APPROVED** [review]
I arrived at the same solution when thinking about this yesterday, so all good. Let's wait for @IngridKJ's reply before merging, though.

**Comment by IngridKJ** [comment]
I performed a quick check with my previously failing grid, and things seem to be working fine now. What I did was simply running my script in the new branch, and I do not get the dimension mismatch I got previously. So, the grid seems to be working fine now.  Are you interested in the details or code for the grid which failed, @keileg?

**Comment by keileg** [comment]
> Are you interested in the details or code for the grid which failed, @keileg?

No, I think we are good. Do you agree, @IvarStefansson?

**Pull request merged**
